api/v1alpha1/hostedcluster_types.go

```diff
@@ -419,10 +419,23 @@ type PlatformSpec struct {
 	// +immutable
 	AWS *AWSPlatformSpec `json:"aws,omitempty"`
 
+	// Agent specifies configuration for agent-based installations.
+	//
+	// +optional
+	// +immutable
+	Agent *AgentPlatformSpec `json:"agent,omitempty"`
+
 	// IBMCloud defines IBMCloud specific settings for components
 	IBMCloud *IBMCloudPlatformSpec `json:"ibmcloud,omitempty"`
 }
 
+// AgentPlatformSpec specifies configuration for agent-based installations.
+type AgentPlatformSpec struct {
+	// AgentNamespace is the namespace where to search for Agents for this cluster
+	// +optional
+	AgentNamespace string `json:"agentNamespace,omitempty"`
```

Comment:
why marked as optional? shouldn't this field be required?

Reply:
Done

Reply:
Can you please remove "omitempty"? then you'll notice that the CRD generated marks the field as required

Reply:
Done